comp/core/tagger/collectors/workloadmeta_main.go

```diff
@@ -76,6 +79,27 @@ func (c *WorkloadMetaCollector) initPodMetaAsTags(labelsAsTags, annotationsAsTag
 	c.nsLabelsAsTags, c.globNsLabels = utils.InitMetadataAsTags(nsLabelsAsTags)
 }
 
+func (c *WorkloadMetaCollector) injectHostTags() {
```

Comment:
this function doesn't follow the normal pattern. 

Any tags that is added by the WorkloadMetaCollector are collected from Workloadmetadastore content.  But this function get directly the data from the `hosttags.Get()` global function. Especially if this data can evolved, in that case you want to be able to update the data.

The correct way to do is:
* add a new entity in workloadmetadatastore that represent the `host` and a new source that represent the `hostmetadata` source : https://github.com/DataDog/datadog-agent/blob/main/comp/core/workloadmeta/types.go 
* Create a new Workloadmetadatastore collector which basically implements what is present  in this `injectHostTags()` function. (here: https://github.com/DataDog/datadog-agent/tree/main/comp/core/workloadmeta/collectors/internal)


the follow will be:
* the new workloadmeta `hostmetadata` collector will create the `host` entity in workloadmeta.
* the creation or update of the `host` entity will generates an event that will populate the tagger.
* in the new workloadmeta `hostmetadata` collector after the `expected_tags_duration` the collector can delete the `host` entity or update a field in the `host` entity to inform the `WorkloadMetaCollector` that the tags attached to the host `workloadmeta-host` can be removed.

Reply:
Thank you for the suggestions. This makes sense to me. Ive made the suggested changes. Let me know what you think